app/src/main/java/net/gsantner/markor/activity/PreviewActivity.java

```diff
@@ -200,11 +201,21 @@ private void shareImage() {
         if (bitmap != null) {
             File image = new File(getExternalCacheDir(), _note.getName() + ".png");
             if (saveBitmap(bitmap, image)) {
-                shareStream(Uri.fromFile(image), "image/png");
+                Uri imageUri = FileProvider.getUriForFile(PreviewActivity.this,
+                        Constants.FILE_PROVIDER_AUTHORITIES,
+                        image);
+                shareStream(imageUri, "image/png");
             }
         }
     }
 
```

Comment:
I would convert this to take parameter (File, MimeType) and call that for image with imagpath and image/png, and for the note with _note and text/plain

Reply:
Thanks. I will do the required changes.

Reply:
Dont feel like you must do things, this project is community work after all 😉

Reply:
haha. okay. You were right actually. It does make code cleaner. I have changed the method signatures. Have a look at it and merge it if it looks alright to you.

Reply:
thx, merging.

Here are more hacktober issues of my projects if you want more:

https://github.com/search?&q=label%3Ahacktoberfest+user%3Agsantner

Maybe look especially into MemeTastic.MemeCreateActivity - if the "share" file is done correctly. It may also use the old method with exposing uri

Reply:
Great! Yea I will look into them too.